tests/testsuite/new.rs

```diff
@@ -455,3 +455,30 @@ fn explicit_project_name() {
         .with_stderr("[CREATED] library `bar` project")
         .run();
 }
+
+#[test]
+fn new_with_edition_2015() {
+    cargo_process("new --edition 2015 foo")
+        .env("USER", "foo")
+        .run();
+    let manifest = fs::read_to_string(paths::root().join("foo/Cargo.toml")).unwrap();
+    assert!(manifest.contains("edition = \"2015\""));
+}
+
+#[test]
+fn new_with_edition_2018() {
+    cargo_process("new --edition 2018 foo")
+        .env("USER", "foo")
+        .run();
+    let manifest = fs::read_to_string(paths::root().join("foo/Cargo.toml")).unwrap();
+    assert!(manifest.contains("edition = \"2018\""));
+}
+
```

Comment:
A 'what is the default edition' test would be good too -- it should be 2018 I believe.

Reply:
Oh, I see that you didn't change the default edition, is there a reason for that? I would semi-expect us to do that.

cc @Centril @aturon

Reply:
Yes agreed.

Reply:
From what I understand, the team discussed this and agreed not to do this early for nightly and just wait for another cycle

Reply:
That's correct: https://paper.dropbox.com/doc/Cargo-team-meeting-notes--AMHODJnMyGYfJPriwBC~8rouAg-Mtg6RFmYjpaKFS1XaTkBh

Reply:
Looking at that document provides no clear rationale to me unless I missed it - the plan is that RC1 is a fully edition 2018 compatible compiler - and making that edition the default for new projects makes sense then.

Could someone from the cargo team perhaps clear up why we didn't want to change the default?

Reply:
The conclusion stemmed from being conservative. The compiler isn't ready right-the-red-hot-second for `--edition 2018` to be the default because there's still unstable features (like the module system) which are in the process of stabilizing. As soon as the compiler is ready we'll change, and we conservatively said that "just before October 25 this must happen"

Reply:
@alexcrichton right but this PR is part of several ones in the queue that will stabilize all the pieces.

Reply:
Ok! It does indeed look like everything is in motion now. I'll make a PR to change the default after this one.

Reply:
I've submitted https://github.com/rust-lang/cargo/pull/5989 which switches to generating 2018 by default.